beniget/beniget.py

```diff
@@ -1248,6 +1280,38 @@ def visit_withitem(self, node):
             self.visit(node.optional_vars)
         return dnode
 
+def _validate_comprehension(node):
+    """
+    Raises SyntaxError if:
+     - a named expression is used in a comprehension iterable expression
+     - a named expression rebinds a comprehension iteration variable
+    """
+    iter_names = set()
+    
+    for gen in node.generators:
+        for name in filter(lambda n: isinstance(n, ast.NamedExpr), ast.walk(gen.iter)):
+            raise SyntaxError('assignment expression cannot be used '
+                                'in a comprehension iterable expression')
+        for name in filter(lambda n: isinstance(n, ast.Name), ast.walk(gen.target)):
+            iter_names.add(name.id)
```

Comment:
could be `iter_names.update(n for n in ast.walk(gen.target) if isinstance(n, ast.Name))`